ocaml-lsp-server/src/hover_req.ml

```diff
@@ -424,12 +326,15 @@ let handle server { HoverParams.textDocument = { uri }; position; _ } mode =
       match Document.kind doc with
       | `Other -> Fiber.return None
       | `Merlin merlin -> (
-        let* parsetree =
+        let* parsetree_and_pipeline =
           Document.Merlin.with_pipeline_exn
             ~name:"hover"
             (Document.merlin_exn doc)
-            (fun pipeline -> Mpipeline.reader_parsetree pipeline)
+            (fun pipeline ->
+              let parsetree = Mpipeline.reader_parsetree pipeline in
+              (parsetree, pipeline))
         in
+        let parsetree, pipeline = parsetree_and_pipeline in
```

Comment:
You can do the split directly on line 329

Reply:
Thank you